scripts/build_sphinx.py

```diff
@@ -0,0 +1,235 @@
+        # detected a sphinx.yml file - this file indicates that
+        # we should include sphinx documentation from external sources.
+
+        # read, parse the config and bring in and convert the necessary
+        # content in the following way:
+        #
+        # - A single sphinx folder is created
+        # - All external repos are cloned into a temp folder and their '/docs'
+        #   folder is copied into the sphinx folders, but named with the name of
+        #   the repo. We end up with a hierarchy of rst files from multiple
+        #   repositories.
+        # - An index.rst is generated in the sphinx folder to include all repos
+        #   so that the doc generation can recurse properly.
+        # - Sphinx is converted into jekyll/markdown for all repos in a single pass.
+        #
+
+        # Example file format:
+        #
+        # repositories:
+        #   Publisher:
+        #       git_url: https://github.com/shotgunsoftware/tk-multi-publish2.git
+        #       parent: Toolkit App Reference
+        #   Core API:
+        #       git_url: https://github.com/shotgunsoftware/tk-core.git
+        #       parent: /
+
+
+        log.info("Detected sphinx.yml configuration file.")
+
+        yaml_file = os.path.join(input_folder, "sphinx.yml")
+        yaml = YAML(typ='safe')
+        with open(yaml_file, "rt") as fh:
+            yaml_data = yaml.load(fh)
+
+        repo_data = yaml_data.get("repositories")
+
+        temp_folder = os.path.join(tempfile.gettempdir(), uuid.uuid4().hex)
+        sphinx_folder = os.path.join(input_folder, "sphinx")
+
+        os.makedirs(temp_folder)
+        os.makedirs(sphinx_folder)
+
+        index_files = []
+
+        # track which repos have a custom parent page
+        custom_parents = {}
+
+        # iterate over repos, clone and move docs into our sphinx folder
+        for (name, params) in repo_data.iteritems():
+
+            (repo_name, _) = os.path.splitext(os.path.basename(params["git_url"]))
+
+            git_folder = os.path.join(temp_folder, repo_name)
+            target_folder = os.path.join(sphinx_folder, repo_name)
+
+            index_files.append("{}/index".format(repo_name))
+
+            execute_external_command(
+                "git clone --depth=50 {repo} {folder}".format(repo=params["git_url"], folder=git_folder)
+            )
+
+            # check out latest tag
+            execute_external_command(
+                "cd {}; git checkout $(git describe --tags $(git rev-list --tags --max-count=1))".format(git_folder)
+            )
+
+            # get latest tag
+            # TODO: Handle injection of version numbers into sphinx structure.
+            # output = execute_external_command("cd {}; git describe --tags".format(git_folder))
+
+            # add python source to pythonpath
+            # so that the sphinx build can auto-document methods etc.
+            add_to_pythonpath(os.path.join(git_folder, "python"))
```

Comment:
I think you're missing [a few additions](https://github.com/shotgunsoftware/tk-internal/blob/master/app_store/python/tank_appstore/sphinx.py#L45-L56) to `PYTHONPATH`, at the very least the `hooks` folder.

Reply:
thanks! added hooks. I think we'll tweak the system as we go too.